pkg/cmd/grafana-cli/utils/grafana_path.go

```diff
@@ -6,35 +6,51 @@ import (
 	"path/filepath"
 
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
+	"golang.org/x/xerrors"
 )
 
 func GetGrafanaPluginDir(currentOS string) string {
 	if isDevEnvironment() {
-		return "../data/plugins"
+		rootPath, err := getGrafanaRoot()
+		if err != nil {
+			panic("Could not get root path. Should not happen after isDevEnvironment() but here we are.")
+		}
+		return filepath.Join(rootPath, "data/plugins")
 	}
 
 	return returnOsDefault(currentOS)
 }
 
-func isDevEnvironment() bool {
-	// if ../conf/defaults.ini exists, grafana is not installed as package
-	// that its in development environment.
+// getGrafanaRoot tries to get root of directory when developing grafana ie repo root. It is not perfect it just
+// checks what is the binary path and tries to guess based on that but if it is not running in dev env you get a bogus
+// path back.
+func getGrafanaRoot() (string, error) {
 	ex, err := os.Executable()
 	if err != nil {
-		logger.Error("Could not get executable path. Assuming non dev environment.")
-		return false
+		return "", xerrors.New("Failed to get executable path")
 	}
 	exPath := filepath.Dir(ex)
 	_, last := path.Split(exPath)
 	if last == "bin" {
 		// In dev env the executable for current platform is created in 'bin/' dir
-		defaultsPath := filepath.Join(exPath, "../conf/defaults.ini")
-		_, err = os.Stat(defaultsPath)
-		return err == nil
+		return filepath.Join(exPath, ".."), nil
 	}
 
 	// But at the same time there are per platform directories that contain the binaries and can also be used.
-	defaultsPath := filepath.Join(exPath, "../../conf/defaults.ini")
+	return filepath.Join(exPath, "../.."), nil
+}
+
+func isDevEnvironment() bool {
```

Comment:
This could maybe be changed to `func tryGetRootForDevEnvironment() (string, bool)`? Then you can skip calling `getGrafanaRoot` one more time after calling this method.

Then you may be able to get rid of the panic above "ould not get root path. Should not happen after isDevEnvironment() but here we are".